File: sdks/csharp/src/NovaBilling/Types/AppliedCouponResponse.cs
using System.Text.Json;
using System.Text.Json.Serialization;
using NovaBilling.Core;

namespace NovaBilling;

[Serializable]
public record AppliedCouponResponse : IJsonOnDeserialized
{
    [JsonExtensionData]
    private readonly IDictionary<string, JsonElement> _extensionData =
        new Dictionary<string, JsonElement>();

    [JsonPropertyName("id")]
    public required string Id { get; set; }

    [JsonPropertyName("couponId")]
    public required string CouponId { get; set; }

    [JsonPropertyName("customerId")]
    public required string CustomerId { get; set; }

    [JsonPropertyName("subscriptionId")]
    public string? SubscriptionId { get; set; }

    [JsonPropertyName("amountOff")]
    public string? AmountOff { get; set; }

    [JsonPropertyName("usesRemaining")]
    public double? UsesRemaining { get; set; }

    [JsonPropertyName("createdAt")]
    public required string CreatedAt { get; set; }

    [JsonIgnore]
    public ReadOnlyAdditionalProperties AdditionalProperties { get; private set; } = new();

    void IJsonOnDeserialized.OnDeserialized() =>
        AdditionalProperties.CopyFromExtensionData(_extensionData);

    /// <inheritdoc />
    public override string ToString()
    {
        return JsonUtils.Serialize(this);
    }
}

File: sdks/ruby/lib/novabilling/types/applied_coupon_response.rb
# frozen_string_literal: true

module Novabilling
  module Types
    class AppliedCouponResponse < Internal::Types::Model
      field :id, -> { String }, optional: false, nullable: false
      field :coupon_id, -> { String }, optional: false, nullable: false, api_name: "couponId"
      field :customer_id, -> { String }, optional: false, nullable: false, api_name: "customerId"
      field :subscription_id, -> { String }, optional: true, nullable: false, api_name: "subscriptionId"
      field :amount_off, -> { String }, optional: true, nullable: false, api_name: "amountOff"
      field :uses_remaining, -> { Integer }, optional: true, nullable: false, api_name: "usesRemaining"
      field :created_at, -> { String }, optional: false, nullable: false, api_name: "createdAt"
    end
  end
end
